api/graphql/notificationQuerySet.js

```diff
@@ -0,0 +1,37 @@
+import presentQuerySet from '../../lib/graphql-bookshelf-bridge/util/presentQuerySet'
+
+const userColumns = q => q.column('users.id', 'name', 'avatar_url')
+
+export default function notificationQuerySet (community, userId, options) {
+  return Notification.query(q => {
+    q.where('reader_id', userId)
+    .leftJoin('activities', function () {
+      this.on('activities.id', '=', 'notifications.activity_id')
+    })
+    q.where('medium', Notification.MEDIUM.InApp)
+    q.limit(options.first)
+    q.offset(options.offset)
+    q.select(bookshelf.knex.raw('notifications.*, count(*) over () as total'))
+    q.orderBy('activities.created_at', 'desc')
+
+    Activity.filterInactiveContent(q)
+    if (community) Activity.joinWithCommunity(community.id, q)
+  })
+}
+
+export function fetchNotificationQuerySet (community, userId, options) {
+  return notificationQuerySet(community, userId, options).fetchAll({withRelated: [
```

Comment:
Wasn't sure about putting all this in the fetchAll here. Am I missing something clever we can do in graphQL?